src/telemetry/metrics/tls_config_reload.rs

```diff
@@ -13,8 +18,22 @@ metrics! {
     }
 }
 
+/// Constructs a Sensor/Fmt pair for TLS config reload metrics.
+pub fn new() -> (Sensor, Fmt) {
+    let inner = Arc::new(RwLock::new(Inner::default()));
+    let fmt = Fmt(Arc::downgrade(&inner));
+    (Sensor(inner), fmt)
+}
+
+/// Supports recording TLS config reload metrics.
+///
+/// When this type is dropped, its metrics may no longer be formatted for prometheus.
+#[derive(Debug)]
+pub struct Sensor(Arc<RwLock<Inner>>);
```

Comment:
Out of interest: do we expect the metrics server to receive enough concurrent requests that it's worth using a `RwLock` rather than a `Mutex`?

Reply:
We don't expect the metrics server to get many requests.

If you're referring to the mutex around `Root`, it's going to be removed in a subsequent PR.

Reply:
Or is your point that RwLock adds overhead over a Mutex?  Does it?

Reply:
> Or is your point that RwLock adds overhead over a Mutex? Does it?
It's somewhat platform-dependent, but at least on Linux, I believe `Mutex` is likely to be faster than `RwLock` when there's a small number of both readers and writers. You may find this interesting: https://www.slideshare.net/mitsunorikomatsu/performance-comparison-of-mutex-rwlock-and-atomic-types-in-rust.

Reply:
That said, I do think that `RwLock` does express intent more clearly, so 🤷‍♀️